merkletree/str.go

```diff
@@ -56,6 +57,8 @@ func (str *SignedTreeRoot) Serialize() []byte {
 	return strBytes
 }
 
+// VerifyHashChain verifies the consistency of the current hash chain,
+// according to section 3.3 per paper.
```

Comment:
As with a few other cases, I would be a bit more specific here about what `VerifyHashChain` actually does. Right now you're just stating the purpose of the function.